### DISCUSSION TITLE
Spaced Repetition System to nudge users to remember their passwords.

### DISCUSSION THREAD

**Discussion by wenman-ualberta** [standardnotes/forum#2791] [Suggestions]
**Have you read the [Longevity Statement](https://standardnotes.org/longevity)?**

Yes. I like it.

**Is your feature request related to a problem? Please describe.**

All software programs have a trade-off between security and convenience. Standard Notes is only different insofar that it weighs heavily towards to security over convenience. But if Standard Notes continues to get more and more popular(Hopefully) the average technical and average security literacy of users is going to be a diminishing line going down and to the right.

This will result in a higher and higher number of customers complaining because they forgot their password and then wondering where the "I forgot my password" link is. Everyone has been conditioned to expect that link. So new users are not going to take the warnings seriously (if they read them) about writing down or memorizing their password, and then if they can't get it reset or their self-made issue fixed, then  they'll get angry, and if they're a paying customer they may stop.

**Describe the solution you'd like**
Solution: Use a [Spaced Repetition System](https://en.wikipedia.org/wiki/Spaced_repetition) to periodically, yet effectively prompt the user the user to enter their password spaced out for minimal input and maximum memorization.

[Note that Signal already does this somewhat, so there is precedence for this.](https://nerdschalk.com/how-to-disable-pin-reminders-on-signal/) 

**Describe alternatives you've considered**

The alternative is to leave it as it is. Obviously this that's fine since this is not high priority as long as the security literacy of new users stays below a level the support person or team can handle.
However if the team finds that the percentage of users who keep on forgetting their password are consuming a significant amount of support $$$ and support team hours this might be a feature to consider.

**Additional context**

This app needs to be as simple as possible and yet also be really easy to use. The relationship between the application and the user should be collaborative and encouraging not punitive and judgemental.

That's all I got. Thoughts?

**Comment by stefan-yas** [comment]
Good food for thought, @A-Casimir-Wenman! Thanks for the kind words as well.

We may consider adding something similar - obviously, we do get "Can you help me recover or reset my password emails?" from time to time. And you are right - appealing to the major audience that potentially does not have good logging, managing and organization habits could lead to frustration amongst users and data-loss.

In most applications, the user has to choose between convenience and security, but, as you said, we do currently lean towards keeping your data secure, even if that means somewhat less convenience, when comparing apps on a user-experience level. Standard Notes' bottom line is bridging the gap between convenience and security, while not compromising on either, but that is a big challenge - technically, architecturally and is a hard thing to do in general.

We make our progress in small increments and with each feature introduction, design or some other change, security is the first thing in our minds, as makers of the product. With each of these milestones, we try to cater to convenience as well - after all, a life-organization and note-taking app should be convenient. In any case, I do like your UX proposal and will bring it up with the team.

Thanks again!
